opendbc/car/toyota/carcontroller.py

```diff
@@ -201,19 +205,23 @@ def update(self, CC, CS, now_nanos):
       if CS.out.standstill or stopping:
         self.filter.x = 0
         self.pid.reset()
+        new_pcm_accel_net = CS.pcm_accel_net
 
       # let PCM handle stopping for now
       pcm_accel_compensation = 0.0
       if not stopping:
         # pcm_accel_compensation = 2.0 * (CS.pcm_accel_net - net_acceleration_request)
+        #pcm_accel_compensation = 2.0 * (new_pcm_accel_net - net_acceleration_request)
         pcm_accel_compensation = 2.0 * (new_pcm_accel_net - net_acceleration_request)
 
       # prevent compensation windup
-      pcm_accel_compensation = clip(pcm_accel_compensation, actuators.accel - self.params.ACCEL_MAX,
-                                    actuators.accel - self.params.ACCEL_MIN)
+      pcm_accel_compensation = clip(pcm_accel_compensation, actuators_accel - self.params.ACCEL_MAX,
+                                    actuators_accel - self.params.ACCEL_MIN)
 
-      self.pcm_accel_compensation = rate_limit(pcm_accel_compensation, self.pcm_accel_compensation, -0.01, 0.01)
-      pcm_accel_cmd = actuators.accel - self.pcm_accel_compensation
+      # TODO: we need a better way to allow the compensation to prevent the request from rising other than insane limits
+      up_limit = 0.1 if CS.out.vEgo < 5 else 0.01
```

Comment:
We need to come up with a better way other than allowing insane limits to allow the request to stay low while we're ramping up to full acceleration. I think a low pass filter w/ no rate limits might do well to filter out the high frequency oscillations we see that this is really here to solve.

Reply:
This is why I needed to write this for testing, the rate limit is too low to prevent the rise of the acceleration request when the PCM compensation sees it overshooting:

![image](https://github.com/user-attachments/assets/7557a72e-248a-40c0-b88a-11367c5e48c2)

Here is it with a PI controller:

...